### PR TITLE
Remove references to dynamic lists on Brexit landing page

### PR DESCRIPTION
**Pull request by laurentqro**
This PR removes a lot of old tech debt referencing `dynamic lists`.

https://trello.com/c/KVmOh3Uk/474-remove-dynamic-lists-tech-debt-from-collections

### CODE REVIEW

File: test/support/transition_landing_page_steps.rb [hunk 1]
```diff
@@ -6,32 +6,15 @@ module TransitionLandingPageSteps
...
   include RummagerHelpers
 
-  def given_there_is_a_transition_taxon
-    stub_content_store_has_item(transition_taxon_path, content_item)
-  end
-
-  def transition_taxon_path
-    "/transition"
-  end
+  CONTENT_ID = "d6c2de5d-ef90-45d1-82d4-5f2438369eea".freeze
```
Comment (vanitabarrett) [inline test/support/transition_landing_page_steps.rb]: What is this the content id of - the taxon? Can we make it clear?

Comment (huwd) [reply test/support/transition_landing_page_steps.rb]: This is the `/transition` taxon (formerly of the parish of `/brexit`)
[Matches here](https://github.com/alphagov/search-api/blob/1f91a38234c2f8c88b43854a3d039d9c26e2a4bf/lib/govuk_index/presenters/common_fields_presenter.rb#L11)

Comment (huwd) [reply test/support/transition_landing_page_steps.rb]: But yeah i agree, could leave these as a comment? Or a link? What's most likely to be updated if we have to change the name again?

Comment (vanitabarrett) [reply test/support/transition_landing_page_steps.rb]: Can we call it `TRANSITION_TAXON_CONTENT_ID`, to make that clearer? Like we call the path `TRANSITION_TAXON_PATH` (below)?

File: test/support/transition_landing_page_steps.rb [hunk 2]
```diff
@@ -6,38 +6,15 @@ module TransitionLandingPageSteps
...
-    stub_content_store_has_item(brexit_taxon_path, content_item)
-  end
-
-  def brexit_taxon_path
-    "/brexit"
-  end
+  CONTENT_ID = "d6c2de5d-ef90-45d1-82d4-5f2438369eea".freeze
+  TRANSITION_TAXON_PATH = "/transition".freeze
 
-  def content_id
```
Comment (huwd) [inline test/support/transition_landing_page_steps.rb]: Or perhaps we resurrect a method like this called `transition_taxon_id` and just shove it in there?

File: test/support/transition_landing_page_steps.rb (hunk 1 above)
Comment (laurentqro) [reply test/support/transition_landing_page_steps.rb]: Good idea, I like `TRANSITION_TAXON_CONTENT_ID`. 👏

**Pull request merged**